File: apps/spring-boot-api/src/main/java/org/bootstrapbugz/api/auth/event/email/ResetPasswordEmail.java
package org.bootstrapbugz.api.auth.event.email;

import com.google.common.io.Files;
import java.io.IOException;
import java.nio.charset.StandardCharsets;
import java.util.Objects;
import lombok.extern.slf4j.Slf4j;
import org.bootstrapbugz.api.shared.email.service.EmailService;
import org.bootstrapbugz.api.user.model.User;
import org.springframework.core.env.Environment;
import org.springframework.core.io.ClassPathResource;

@Slf4j
public class ResetPasswordEmail implements JwtEmail {
  @Override
  public void sendEmail(
      EmailService emailService, Environment environment, User user, String token) {
    try {
      final var template = new ClassPathResource("templates/email/reset-password.html").getFile();
      final String link = environment.getProperty("ui.app.url") + "/reset-password?token=" + token;
      final String body =
          Files.asCharSource(template, StandardCharsets.UTF_8)
              .read()
              .replace("$name", user.getUsername())
              .replace("$link", link)
              .replace("$appName", Objects.requireNonNull(environment.getProperty("app.name")));
      emailService.sendHtmlEmail(user.getEmail(), "Reset Password", body);
    } catch (IOException e) {
      log.error(e.getMessage());
    }
  }
}

File: backend/spring-boot-kotlin/src/main/java/org/bootstrapbugz/api/auth/jwt/event/email/ResetPasswordEmail.kt
package org.bootstrapbugz.api.auth.jwt.event.email

import org.bootstrapbugz.api.shared.email.service.EmailService
import org.bootstrapbugz.api.user.model.User
import org.slf4j.LoggerFactory
import org.springframework.core.env.Environment
import org.springframework.core.io.ClassPathResource
import java.io.IOException
import java.nio.charset.StandardCharsets
import java.util.*

class ResetPasswordEmail : JwtEmail {
    companion object {
        private val log = LoggerFactory.getLogger(ResetPasswordEmail::class.java)
    }

    override fun sendEmail(emailService: EmailService, environment: Environment, user: User, token: String) {
        try {
            val templateFile = ClassPathResource("templates/email/reset-password.html").file
            val link = "${environment.getProperty("ui.app.url")}/auth/reset-password?token=$token"
            val body = templateFile.readText(StandardCharsets.UTF_8)
                .replace("\$name", user.username)
                .replace("\$link", link)
                .replace("\$appName", Objects.requireNonNull(environment.getProperty("app.name")))
            emailService.sendHtmlEmail(user.email, "Reset password", body)
        } catch (e: IOException) {
            log.error(e.message ?: "An error occurred while sending reset password email", e)
        }
    }
}
